File: Server/Domain/Entities/Work.cs
using Domain.Enums;

namespace Domain.Entities
{
    public class Work : BaseEntity
    {
        public required string Title { get; set; }
        public DateOnly? TimePublished { get; set; }
        public int? TotalAuthors { get; set; }
        public int? TotalMainAuthors { get; set; }
        public Dictionary<string, string>? Details { get; set; }
        public WorkSource Source { get; set; }
        public bool IsLocked { get; set; }

        public Guid WorkTypeId { get; set; }
        public Guid? WorkLevelId { get; set; }
        public Guid SystemConfigId { get; set; }

        public virtual WorkType? WorkType { get; set; }
        public virtual WorkLevel? WorkLevel { get; set; }
        public virtual SystemConfig? SystemConfig { get; set; }
        public virtual ICollection<Author>? Authors { get; set; } = new List<Author>();
        public virtual ICollection<WorkAuthor>? WorkAuthors { get; set; } = new List<WorkAuthor>();

        public DateOnly? ExchangeDeadline { get; set; }
    }
}

File: Client/src/lib/types/models/Work.ts
import { WorkSource } from "../enums/WorkSource";
import { Author, UpdateAuthorRequest } from "./Author";

export interface Work {
  id: string;
  title: string;
  timePublished?: string;
  totalAuthors?: number;
  totalMainAuthors?: number;
  details?: Record<string, string>;
  source: WorkSource;
  isLocked: boolean;
  workTypeId: string;
  workLevelId?: string;
  workTypeName?: string;
  workLevelName?: string;

  academicYearId?: string;
  academicYearName?: string;
  exchangeDeadline?: string;


  createdDate: string;
  modifiedDate: string | null;
  coAuthorUserIds: string[];
  author?: {
    authorRoleId: string;
    purposeId: string;
    position?: number;
    scoreLevel?: number;
    scImagoFieldId?: string;
    fieldId?: string;
  };
  authors?: Author[];
}

export interface CreateWorkRequest {
  title: string;
  timePublished?: string;
  totalAuthors?: number;
  totalMainAuthors?: number;
  details?: Record<string, string>;
  source: number;
  workTypeId: string;
  workLevelId?: string;
  academicYearId?: string;
  author: {
    authorRoleId: string;
    purposeId: string;
    position?: number;
    scoreLevel?: number;
    scImagoFieldId?: string;
    fieldId?: string;
  };
  coAuthorUserIds: string[];
}

export interface UpdateWorkRequest {
  title?: string;
  timePublished?: string;
  totalAuthors?: number;
  totalMainAuthors?: number;
  details?: Record<string, string>;
  source: number;
  workTypeId?: string;
  workLevelId?: string;
  academicYearId?: string;
  coAuthorUserIds: string[];
}

export interface UpdateWorkWithAuthorRequest {
  workRequest?: UpdateWorkRequest;
  authorRequest?: UpdateAuthorRequest;
}
